config/settings/test.yml

```diff
@@ -107,14 +107,11 @@ bgs:
   client_station_id: 281
   client_username: VAgovAPI
   url: https://internal-dsva-vagov-dev-fwdproxy-1893365470.us-gov-west-1.elb.amazonaws.com:4447
-  # To create a new VCR cassette:
-  # 1. Uncomment these lines
-  # 2. Create an empty cassette file
-  # 3. Run the test against that file, VCR should fill it
-  # 4. Replace localhost in the file with the url above
-  # 5. Re-comment these lines:
-  # url: https://localhost:4447
-  # ssl_verify_mode: "none"
+# Put this in `config/settings/test.local.yml` and then erb-template in
+# `Settings.bgs.url` to cassettes that are run with the VCR option `erb: true`
+# bgs:
+#   url: https://localhost:4447
+#   ssl_verify_mode: "none"
```

Comment:
trying to follow this? we'll have to add `erb: true` to the specs to get this? plus add or create this `test.local.yml`? what's the value in that vs current state? besides the fact you probably won't have to remove `erb: true` after running it?

Reply:
`BGSClientHelper` exposes `use_bgs_cassette` which amounts to a small bundle of QoL improvements for testing new BGS operation wrapper implementations. One of the QoL improvements is that the cassettes [can template in values](https://github.com/department-of-veterans-affairs/vets-api/blob/77954a8dbbb25d4933682704de32ccb7019c9c6a/modules/claims_api/spec/support/bgs_client_helpers.rb#L24). The convenient part of that is that there is no longer a need to manipulate the URL of a cassette request past the first time where the dev makes it look [like this](https://github.com/department-of-veterans-affairs/vets-api/blob/77954a8dbbb25d4933682704de32ccb7019c9c6a/spec/support/vcr_cassettes/bgs/manage_representative_service/read_poa_request/invalid_external_uid_and_key.yml#L5). The current state is inconvenient if you have change the URL so the cassette works in CI and then suppose you come back and do more work on that test example and you have to update the URL again for localhost. So current state implies updating URL value every time test running moves from CI to localhost or vice versa. New state would be template value in once and never update again.

Reply:
With that said, I don't particularly understand the purpose behind the value we use in CI.